src/apps/applets/crud/applets.py

```diff
@@ -63,6 +63,16 @@ async def update_by_id(
         )
         return instance
 
+    async def get_by_link(
+        self, link: uuid.UUID, require_login: bool
+    ) -> AppletSchema | None:
+        query: Query = select(AppletSchema)
+        query = query.where(AppletSchema.link == link)
+        query = query.where(AppletSchema.require_login == require_login)
+
+        db_result = await self._execute(query)
+        return db_result.scalars().first()
```

Comment:
Don't you think it is better to raise NotFoundError instead of returning the None object?

Reply:
No, because it is very low level api, so you can use it in several places which might return success or failure

Reply:
If it will be handled in service level, then it is good

Reply:
As you wish